iml-gui/crate/src/page/filesystems.rs

```diff
@@ -1,57 +1,130 @@
-    let fs = &model.records.filesystem;
-    if fs.is_empty() {
+pub fn view(cache: &ArcCache, model: &Model, all_locks: &Locks) -> impl View<Msg> {
+    if cache.filesystem.is_empty() {
         div![
             class![C.text_3xl, C.text_center],
-            h1![class![C.m_2, C.text_gray_600], "No file systems are configured"],
+            h1![class![C.m_2, C.text_gray_600], "No filesystems found"],
         ]
     } else {
         div![
             class![C.bg_white, C.border_t, C.border_b, C.border, C.rounded_lg, C.shadow],
             div![
                 class![C.flex, C.justify_between, C.px_6, C._mb_px, C.bg_gray_200],
-                h3![class![C.py_4, C.font_normal, C.text_lg], "File systems"]
+                h3![class![C.py_4, C.font_normal, C.text_lg], "Filesystems"]
             ],
             T::wrapper_view(vec![
                 T::thead_view(vec![
-                    T::th_view(plain!("File System")),
-                    T::th_view(plain!("Management Server")),
-                    T::th_view(plain!("Metadata Target Count")),
-                    T::th_view(plain!("Connected Clients")),
-                    T::th_view(plain!("Space Used / Total")),
+                    T::th_view(plain!["Filesystem"]),
+                    T::th_view(plain!["Primary MGS"]),
+                    T::th_view(plain!["MDT Count"]),
+                    T::th_view(plain!["Connected Clients"]),
+                    T::th_view(plain!["Space Used / Total"]),
                 ]),
-                tbody![fs.arc_values().map(|f| tr![
-                    T::td_view(vec![
-                        fs_link(f),
-                        span![class![C.mx_1], lock_indicator::view(&model.locks, f)],
-                        alert_indicator(&model.records.active_alert, f, true, Placement::Top)
-                    ]),
-                    T::td_view(filesystem::mgs(&model.records.target, f)),
-                    T::td_right(plain!(f.mdts.len().to_string())),
-                    T::td_right(filesystem::clients_view(f)),
-                    T::td_view(filesystem::size_view(f)),
-                    td!["TBD"]
-                ],)],
-            ],)
+                tbody![model.filesystems.iter().map(|f| match model.rows.get(&f.id) {
+                    None => empty![],
+                    Some(row) => tr![
+                        T::td_view(vec![
+                            fs_link(f),
+                            lock_indicator::view(all_locks, f).merge_attrs(class![C.mr_2]),
+                            alert_indicator(&cache.active_alert, f, true, Placement::Right)
+                        ])
+                        .merge_attrs(class![C.text_center]),
+                        T::td_view(filesystem::mgs(&cache.target.arc_values().cloned().collect(), f))
+                            .merge_attrs(class![C.text_center]),
+                        T::td_right(plain![f.mdts.len().to_string()]).merge_attrs(class![C.text_center]),
```

Comment:
The content is a number, I think it is better be aligned to the right. On the other side, I can't imagine someone want to compare these number for different filesystems.

P. S. We definitely need `td_center` :)